lib/view/monthlyTaskView.dart

```diff
@@ -1,60 +1,96 @@
 import 'package:flutter/material.dart';
+import 'package:get/get.dart';
 import 'package:planner/common/database.dart';
-import 'package:planner/common/time_management.dart';
 import 'package:planner/models/task.dart';
 import 'package:table_calendar/table_calendar.dart';
 import 'package:planner/view/taskView.dart';
 
 class MonthlyTaskView extends StatefulWidget {
-  const MonthlyTaskView({super.key});
-
   @override
   _MonthlyTaskViewState createState() => _MonthlyTaskViewState();
 }
 
 class _MonthlyTaskViewState extends State<MonthlyTaskView> {
-  CalendarFormat _calendarFormat = CalendarFormat.month;
+  final CalendarFormat _calendarFormat = CalendarFormat.month;
   DateTime _focusedDay = DateTime.now();
   DateTime? _selectedDay;
   DatabaseService db = DatabaseService();
   List<Task> todayTasks = [];
-
-  // Add a PageController for handling page navigation
-   final PageController _pageController = PageController();
+  final PageController _pageController = PageController();
+  Map<DateTime, List<Task>> active = {};
 
   @override
   void initState() {
     super.initState();
-    // Add a listener to the PageController to update the focusedDay
-    _pageController.addListener(() {
-      setState(() {
-        _focusedDay = _pageController.page == 0
-            ? _focusedDay = getDateOnly(_focusedDay, offsetMonths: -1)
-            : _focusedDay = getDateOnly(_focusedDay, offsetMonths: 1);
-      });
-    });
+    fetchMonthlyTasks(DateTime.now());
   }
 
   @override
   void dispose() {
-    // Dispose of the PageController to prevent memory leaks
     _pageController.dispose();
     super.dispose();
   }
 
+  bool hasTasks(
+      DateTime date,
+      Map<DateTime, List<Task>> activeMap,
+      Map<DateTime, List<Task>> delayedMap,
+      Map<DateTime, List<Task>> completedMap) {
+    List<Task> tasks = [
+      ...?activeMap[date],
+      ...?delayedMap[date],
+      ...?completedMap[date]
+    ];
+    return tasks.isNotEmpty;
+  }
+
+  // This function returns a new DateTime object with only year, month, and day
+  DateTime _toDay(DateTime dateTime) {
+    return DateTime(dateTime.year, dateTime.month, dateTime.day);
+  }
+
+// Use this function when you're setting and getting tasks from the active map
+  void fetchMonthlyTasks(DateTime selectedDate) async {
+    DateTime dateStart = _toDay(selectedDate);
+    Map<DateTime, List<Task>> activeMap, delayedMap, completedMap;
+    (activeMap, delayedMap, completedMap) =
+        await db.getTaskMapsMonth(dateStart);
+
+    active = activeMap.map((key, value) => MapEntry(
+        _toDay(key), value)); // Use _toDay when setting tasks in the active map
+    todayTasks = active[_toDay(selectedDate)] ??
+        []; // Use _toDay when getting tasks from the active map
+
+    setState(() {});
+    print(todayTasks);
+  }
+
   void fetchTodayTasks(DateTime selectedDate) async {
```

Comment:
Fucntions like these shouldn't be in this file/parent folder. @anlhu We should probably move all database related functions from here into the database file.

Reply:
hmm, maybe not the setState part. should the function just be renamed into setMonthlyState? not sure what its doing